src/core/core_utils.js

```diff
@@ -329,15 +330,15 @@ function _collectJS(entry, xref, list, parents) {
       _collectJS(element, xref, list, parents);
     }
   } else if (entry instanceof Dict) {
-    if (isName(entry.get("S"), "JavaScript") && entry.has("JS")) {
+    if (isName(entry.get("S"), "JavaScript")) {
       const js = entry.get("JS");
       let code;
-      if (isStream(js)) {
+      if (js instanceof BaseStream) {
         code = js.getString();
-      } else {
+      } else if (typeof js === "string") {
         code = js;
       }
-      code = stringToPDFString(code);
+      code = code && stringToPDFString(code);
```

Comment:
nano-nit: use `&&=` to shorten the line

Reply:
I'd love to do that, but unfortunately we're *still* blocked on [bug 1247687](https://bugzilla.mozilla.org/show_bug.cgi?id=1247687); please see PR #12563.